src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -105,6 +125,60 @@ public HubConnection(IConnectionFactory connectionFactory, IHubProtocol protocol
 
             _loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
             _logger = _loggerFactory.CreateLogger<HubConnection>();
+
+            // closes this class, if it goes `15s` without hearing from server
+            // starts in receive loop
+            _timeoutTimer = new PeriodicConnectionTimer(
```

Comment:
Any reason we aren't using timer awaitable? I dislike the fact that we have these callbacks that are in the middle of the constructor.